features/user/components/userProfile/EditProfilePanel.tsx

```diff
@@ -0,0 +1,146 @@
+'use client'
+
+import { Card, CardHeader, CardContent } from '@/components/ui/card'
+import {
+	Form,
+	FormControl,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useForm } from 'react-hook-form'
+import { EditProfileSchema } from '../../schemas'
+import { z } from 'zod'
+import { Button } from '@/components/ui/button'
+import { signIn, useSession } from 'next-auth/react'
+import { useEffect, useState, useTransition } from 'react'
+import { editUser, getUserById } from '../../actions'
+import { User } from '../../types'
+import { LoadingSpinner } from '@/components/ui/LoadingSpinner'
+import { useToast } from '@/components/ui/use-toast'
+
+type EditProfileFormData = z.infer<typeof EditProfileSchema>
+
+export const EditProfilePanel = () => {
+	const [user, setUser] = useState<User | null>()
+	const [isFetchUserPending, startFetchUserTransition] = useTransition()
+
+	const { toast } = useToast()
+
+	const form = useForm<EditProfileFormData>({
+		resolver: zodResolver(EditProfileSchema),
+	})
+	const {
+		handleSubmit,
+		formState: { errors },
+	} = form
+
+	const { data: session, update } = useSession()
+	const currentUserId = session?.user?.id
+
+	useEffect(() => {
+		const fetchUser = async () => {
+			if (!currentUserId) {
+				return
+			}
+			const user = await getUserById(currentUserId)
+			setUser(user)
+		}
+		startFetchUserTransition(() => {
+			fetchUser().then().catch()
+		})
+	}, [currentUserId])
+
+	const onSubmit = async (data: EditProfileFormData) => {
+		console.log(data)
+		if (!currentUserId) return
+		editUser({ name: data.name, image: data.image || '' }, currentUserId)
+			.then(resData => {
+				toast({
+					title: 'Success!!',
+					description: resData || 'Profile updated successfully',
+				})
+				update()
```

Comment:
Nie wiadomo co update. Widzę że pobierasz z useSession() to pewnie usera/sesje. Przy destrukturyzacji daj dla update jakiś aliast np. updateSession() czy updateUser()